tests/src/test/java/org/mozilla/javascript/tests/DefaultParametersTest.java

```diff
@@ -0,0 +1,383 @@
+        final String script = "function f({ z = 3, x = 2 } = {}) {\n" + "  return z;\n" + "}\n";
+        assertIntEvaluates(3, script + "f()");
+        assertIntEvaluates(3, script + "f({})");
+        assertIntEvaluates(2, script + "f({z: 2})");
+    }
+
+    @Test
+    public void defaultParametersWithArgumentsObject() throws Exception {
+        final String script =
+                "function f(a = 55) {\n"
+                        + "  arguments[0] = 99; // updating arguments[0] does not also update a\n"
+                        + "  return a;\n"
+                        + "}\n"
+                        + "\n"
+                        + "function g(a = 55) {\n"
+                        + "  a = 99; // updating a does not also update arguments[0]\n"
+                        + "  return arguments[0];\n"
+                        + "}\n"
+                        + "\n"
+                        + "// An untracked default parameter\n"
+                        + "function h(a = 55) {\n"
+                        + "  return arguments.length;\n"
+                        + "}\n";
+        assertIntEvaluates(10, script + "f(10)");
+        assertIntEvaluates(55, script + "f()");
+        assertIntEvaluates(10, script + "g(10)");
+        assertEvaluates(Undefined.instance, script + "g()");
+        assertIntEvaluates(0, script + "h()");
+        assertIntEvaluates(1, script + "h(10)");
+    }
+
+    @Test
+    public void functionDefaultArgsArray() throws Exception {
+        final String script =
+                "function append(value, array = []) {\n"
+                        + "  array.push(value);\n"
+                        + "  return array;\n"
+                        + "}\n"
+                        + "\n";
+        assertIntEvaluates(1, script + "append(1)[0]");
+        assertIntEvaluates(2, script + "append(2)[0]");
+    }
+
+    @Test
+    public void functionDefaultArgsObject() throws Exception {
+        final String script =
+                "function append(key, value, obj = {}) {\n"
+                        + "  obj[key]=value;\n"
+                        + "  return obj;\n"
+                        + "}\n"
+                        + "\n";
+        assertIntEvaluates(1, script + "append('a', 1)['a']");
+        assertIntEvaluates(2, script + "append('a', 2)['a']");
+    }
+
+    private static void assertThrows(final String expected, final String source) {
+        assertThrowsWithLanguageLevel(expected, source, Context.VERSION_ES6);
+    }
+
+    private static void assertThrowsWithLanguageLevel(
+            String expected, final String source, int languageLevel) {
+        Utils.runWithAllOptimizationLevels(
+                cx -> {
+                    int oldVersion = cx.getLanguageVersion();
+                    cx.setLanguageVersion(languageLevel);
+                    try {
+                        final Scriptable scope = cx.initStandardObjects();
+                        Object rep = null;
+                        try {
```

Comment:
You can use `var error = assertThrows(EcmaError.class, () -> cx.evaluateString(...))` to avoid the try/catch